components/CosmWasmQuery.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from "react";
+import {
+  getCosmosChainFromConfig,
+  useGlobalContext,
+} from "../contexts/GlobalContext";
+import CosmosNetworkSelector from "./CosmosNetworkSelector";
+import { CosmWasmClient, JsonObject } from "@cosmjs/cosmwasm-stargate";
+
+interface CosmWasmQueryProps {
+  buildQuery: (kvs: Record<string, string>) => JsonObject | undefined;
+}
+
+/**
+ * Allow the user to send a query message to a cosmos chain and visualize the response.
+ * This component will invoke `buildMsg` on the global context to create the message.
+ * `buildMsg` may return `undefined` to indicate that the key-value store does not contain all
+ * of the necessary arguments.
+ */
+const CosmWasmQuery = ({ buildQuery }: CosmWasmQueryProps) => {
+  const [response, setResponse] = useState<string>();
+  // The preface is explainer text that shows up before the response itself.
+  const [responsePreface, setResponsePreface] = useState<string>();
+
+  const { keyValueStore, cosmosChainConfig } = useGlobalContext();
+
+  useEffect(() => {
+    clearResponse();
+  }, [keyValueStore, cosmosChainConfig]);
+
+  const sendTransaction = async () => {
+    const queryJson: JsonObject | undefined = buildQuery(keyValueStore);
+
+    const chain = getCosmosChainFromConfig(cosmosChainConfig.chainId);
```

Comment:
this function call here takes awhile to load and in the meanwhile nothing is showed in the responsePreface which is not an ideal UX, what do you think about moving `setResponsePreface("Loading...");` up here?